src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go

```diff
@@ -222,15 +285,15 @@ func initConfig() {
 			},
 		},
 		ScalingEngine: config.ScalingEngineConfig{
-			ScalingEngineURL: scalingEngine.URL(),
+			ScalingEngineURL: mockScalingEngine.URL(),
 			TLSClientCerts: models.TLSCerts{
 				KeyFile:    filepath.Join(testCertDir, "eventgenerator.key"),
 				CertFile:   filepath.Join(testCertDir, "eventgenerator.crt"),
 				CACertFile: filepath.Join(testCertDir, "autoscaler-ca.crt"),
 			},
 		},
 		MetricCollector: config.MetricCollectorConfig{
-			MetricCollectorURL: metricCollector.URL(),
+			MetricCollectorURL: mockLogCacheEndpoint,
```

Comment:
it would be nice if the MockLogCache had an .URL() method too. Is a bit strange that the Start returns the url vs returning an err or nil.

Reply:
makes sense, then I also can get rid of these variables that store the port etc. :D 👍 

[16f04ba](https://github.com/cloudfoundry/app-autoscaler-release/pull/2983/commits/16f04bab716722592a41e2398c999083f87d9de5)